### PR TITLE
Update resnet.py with New Numpy Operators

### PR DESCRIPTION
**Pull request by jinboci**
Update resnet.py with new NumPy operators

### CODE REVIEW

File: gluoncv/model_zoo/resnet.py [hunk 1]
```diff
@@ -31,10 +31,13 @@
            'se_resnet101_v2', 'se_resnet152_v2',
            'get_resnet']
 
+import mxnet as mx
 from mxnet.context import cpu
 from mxnet.gluon.block import HybridBlock
 from mxnet.gluon import nn
 from mxnet.gluon.nn import BatchNorm
+from mxnet import use_np
+mx.npx.set_np()
```
Comment (zhreshold) [inline gluoncv/model_zoo/resnet.py:40]: the numpy mode switch is perhaps not ready for master. I have created a branch: https://github.com/dmlc/gluon-cv/tree/numpy for numpy transition.

File: gluoncv/model_zoo/resnet.py [hunk 2]
```diff
@@ -106,18 +110,18 @@ def hybrid_forward(self, F, x):
         x = self.body(x)
 
         if self.se:
-            w = F.contrib.AdaptiveAvgPooling2D(x, output_size=1)
+            w = F.contrib.AdaptiveAvgPooling2D(x.as_nd_ndarray(), output_size=1).as_np_ndarray()
```
Comment (zhreshold) [inline gluoncv/model_zoo/resnet.py:113]: this is fine for a temporary solution, but I am expecting the operator will be available in `npx`

File: gluoncv/model_zoo/resnet.py [hunk 3]
```diff
@@ -106,18 +110,18 @@ def hybrid_forward(self, F, x):
...
             w = self.se(w)
-            x = F.broadcast_mul(x, w.expand_dims(axis=2).expand_dims(axis=2))
+            x = x * w.as_nd_ndarray().expand_dims(axis=2).expand_dims(axis=2).as_np_ndarray()
```
Comment (zhreshold) [inline gluoncv/model_zoo/resnet.py:115]: similar here

File: gluoncv/model_zoo/resnet.py (hunk 1 above)
Comment (jinboci) [reply gluoncv/model_zoo/resnet.py:40]: Thank you for your review. I have changed the base to the NumPy branch.

**Comment by zhreshold** [comment]
There's an error in unittest:

```
Traceback (most recent call last):

  File "/var/lib/jenkins/miniconda3/envs/gluon_cv_py3_test/lib/python3.6/site-packages/nose/case.py", line 197, in runTest

    self.test(*self.arg)

  File "/var/lib/jenkins/workspace/workspace/gluon-cv-py3/tests/unittests/test_data_dataloader.py", line 85, in test_random_transform_dataloader

    for batch in loader:

  File "/var/lib/jenkins/miniconda3/envs/gluon_cv_py3_test/lib/python3.6/site-packages/mxnet/gluon/data/dataloader.py", line 473, in __next__

    batch = pickle.loads(ret.get(self._timeout))

  File "/var/lib/jenkins/miniconda3/envs/gluon_cv_py3_test/lib/python3.6/multiprocessing/pool.py", line 644, in get

    raise self._value

TypeError: Operator `_npi_stack` registered in backend is known as `stack` in Python. This is a numpy operator which can only accept MXNet numpy ndarrays, while received a legacy ndarray. Please ensure that you have activated numpy semantics by calling `npx.set_np()` in your code. If you still see this error with numpy semantics activated, please call `as_np_ndarray()` upon the legacy ndarray to convert it to an MXNet numpy ndarray, and then feed the converted array to this operator.
```

**Comment by zhreshold** [comment]
@jinboci I suggest you can disable all unrelated unittest by
```
import unittest


@unittest.skip("temporarily disabled")
def test_case():
   ...
```

or only collect the unttest you are trying to test in the numpy branch

**Comment by mli** [comment]
Job PR-1408-55 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1408/55/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1408-57 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1408/57/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1408-59 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1408/59/index.html 
Code coverage of this PR:  vs. Master:

**Comment by jinboci** [comment]
@zhreshold Thank you very much for your review. Most of the test cases are disabled except for those directly testing resnet.py. Also, image.py is modified to pass the tests in test_data_tansforms.py

**Comment by zhreshold** [comment]
@jinboci Nice job on refactoring this, and thanks for putting efforts on initiating the process, I will merge this to unblock the parallel efforts

**Pull request merged**